linera-core/src/updater.rs

```diff
@@ -151,6 +152,13 @@ where
                 highest_key_score = highest_key_score.max(entry.0);
             }
             Err(err) => {
+                // TODO(#2857): Handle non-remote errors properly.
+                let err = match err {
+                    ChainClientError::RemoteNodeError(err) => err,
+                    err => NodeError::LocalError {
+                        error: err.to_string(),
```

Comment:
`NodeError` means that a remote node caused the failure?

Reply:
Yes, that's why I added the `TODO`.
(We already converted everything to a `NodeError` before this change, but it was all over the place. Now we only do it here, right after the `TODO`, and at least the other functions can already return the correct error types.)

Reply:
I'm not sure how to read that even 🤔 `RemoteNodeError`s were supposed to mean a failure caused by the remote node but `LocalError` says it's error when handling a response from the remote node. So who's fault it is?

Can we rename `NodeError::LocalError` (which will be `RemoteNodeError::LocalError`) to something else - not sure what it should be though?

Reply:
Could `LocalError` be renamed to `ResponseHandlingError`?